src/toil/jobStores/aws/jobStore.py

```diff
@@ -299,6 +303,65 @@ def deleteFile(self, jobStoreFileID):
         else:
             info.delete()
 
+    def importFile(self, sourceUrl):
+        url = urlparse.urlparse(sourceUrl)
+        if url.scheme.lower() == 's3':
+            if len(sourceUrl.split('/')) != 4:
+                raise RuntimeError("The url '%s' is not a valid s3 url" % sourceUrl)
+            srcBucket = self._loadBucketOrFail(url.netloc)
+
+            try:
+                srcKey = srcBucket.get_key((url.path)[1:])
+                assert srcKey is not None
+            except AssertionError:
+                raise RuntimeError("key '%s' does not exists in bucket '%s'" % ((url.path)[1:], url.netloc))
```

Comment:
I would refactor this into an if statement that checks if the key is none. try/except can be more efficient than if statements in cases like 
```python
if key in dictionary:
    return dictionary[key]
```
where we are doing a redundant key lookup, but in this case the if statement would be just as efficient as well as more concise.